silhouette/test/com/mohiva/play/silhouette/impl/providers/BasicAuthProviderSpec.scala

```diff
@@ -135,15 +150,29 @@ class BasicAuthProviderSpec extends PlaySpecification with Mockito {
     lazy val fooHasher = {
       val h = mock[PasswordHasher]
       h.id returns "foo"
+      h.isSuitable(any) answers { p =>
+        p match {
+          case PasswordInfo("foo", _, _) => true
+          case _                         => false
+        }
+      }
+      h.isDeprecated(any) returns Some(false)
       h
     }
 
     /**
-     * An optional password hasher.
+     * A deprecated password hasher.
      */
     lazy val barHasher = {
       val h = mock[PasswordHasher]
       h.id returns "bar"
+      h.isSuitable(any) answers { p =>
```

Comment:
The only difference between the two hashers is the `id`, so can't you factor out a function?